pkg/cloudprovider/providers/aws/aws.go

```diff
@@ -298,6 +298,12 @@ type Cloud struct {
 	mutex                    sync.Mutex
 	lastNodeNames            sets.String
 	lastInstancesByNodeNames []*ec2.Instance
+
+	// We keep an active list of devices we have assigned but not yet
+	// attached, to avoid a race condition where we assign a device mapping
+	// and then get a second request before we attach the volume
+	attachingMutex sync.Mutex
+	attaching      map[ /*nodeName*/ string]map[mountDevice]string
```

Comment:
Actually it is just attach operations.  When called in detach, it is called with false so we won't add a map entry.  The confusion is that a Detach operation will cancel an in-flight Attach operation.